src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultProjectConfigurationFilePathStore.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Composition;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
+    [Export(typeof(ProjectConfigurationFilePathStore))]
+    internal class DefaultProjectConfigurationFilePathStore : ProjectConfigurationFilePathStore
+    {
+        private readonly Dictionary<string, string> _mappings;
+        private readonly object _mappingsLock;
+
+        public override event EventHandler<ProjectConfigurationFilePathChangedEventArgs> Changed;
+
+        [ImportingConstructor]
+        public DefaultProjectConfigurationFilePathStore()
+        {
+            _mappings = new Dictionary<string, string>(FilePathComparer.Instance);
+            _mappingsLock = new object();
+        }
+
+        public override IReadOnlyDictionary<string, string> GetMappings() => new Dictionary<string, string>(_mappings);
+
+        public override void Set(string projectFilePath, string configurationFilePath)
```

Comment:
I remember an offline conversation where you said there could be multiple files providing tag helpers per project. How will that work with this? Or am I completely misremembering?

Reply:
Hmm so there's an interesting case where you have a debug and release directory which both contain a `project.razor.json`. I'm not sure how that's handled here to be honest. I'll have to give that a try

Reply:
lol soooo this "just works". Lets see if I can explain. So in VS there's an active configuration dialog: 
![image](https://user-images.githubusercontent.com/2008729/87097542-50372f00-c1fa-11ea-930f-9f037f80e76d.png)

When you switch that from Debug -> Release or vice versa it triggers a project update which will update the project file path to have the correct `project.razor.json` publish directory whether it be the Debug or Release version.

Now that being said in the case where everything is in the workspace directory including the Release/Debug folders you'll have 2 project.razor.json files in that scope. However, our language server only operates on "last updated" project.razor.json's. Meaning when a project loads whatever scope is active updates the "last updated" project.razor.json resulting in the language server always operating on the correct config.

Reply:
Interesting. Where is our code that only looks at the last updated file?

Reply:
It's our `ConfigurationFileChangeDetector`: https://github.com/dotnet/aspnetcore-tooling/blob/f4073da6dc2e7bdda6f5a29f53b7138ead540bf5/src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectConfigurationFileChangeDetector.cs#L83

Basically it's listening to file change events on disk